satpy/tests/reader_tests/test_ahi_gridded.py

```diff
@@ -0,0 +1,262 @@
+
+import unittest
+from unittest import mock
+import numpy as np
+import dask.array as da
+from pyresample.geometry import AreaDefinition
+from satpy.readers.ahi_gridded import AHIGriddedFileHandler
+
+
+class TestAHIGriddedArea(unittest.TestCase):
+    """Test the AHI gridded reader definition."""
+
+    def setUp(self):
+        self.FULLDISK_SIZES = {0.005: {'x_size': 24000,
+                                       'y_size': 24000},
+                               0.01: {'x_size': 12000,
+                                      'y_size': 12000},
+                               0.02: {'x_size': 6000,
+                                      'y_size': 6000}}
+
+        self.AHI_FULLDISK_EXTENT = [85., -60., 205., 60.]
+
+    @staticmethod
+    def make_fh(filetype, area='fld'):
+        """Create a test file handler."""
+        m = mock.mock_open()
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            fh = AHIGriddedFileHandler('somefile',
+                                       {'area': area},
+                                       filetype_info={'file_type': filetype})
+            return fh
+
+    def test_low_res(self):
+        """Check size of the low resolution (2km) grid."""
+        tmp_fh = self.make_fh('tir.01')
+        self.assertEqual(self.FULLDISK_SIZES[0.02]['x_size'], tmp_fh.ncols)
+        self.assertEqual(self.FULLDISK_SIZES[0.02]['y_size'], tmp_fh.nlines)
+
+    def test_med_res(self):
+        """Check size of the low resolution (1km) grid."""
+        tmp_fh = self.make_fh('vis.02')
+        self.assertEqual(self.FULLDISK_SIZES[0.01]['x_size'], tmp_fh.ncols)
+        self.assertEqual(self.FULLDISK_SIZES[0.01]['y_size'], tmp_fh.nlines)
+
+    def test_hi_res(self):
+        """Check size of the low resolution (0.5km) grid."""
+        tmp_fh = self.make_fh('ext.01')
+        self.assertEqual(self.FULLDISK_SIZES[0.005]['x_size'], tmp_fh.ncols)
+        self.assertEqual(self.FULLDISK_SIZES[0.005]['y_size'], tmp_fh.nlines)
+
+    def test_area_def(self):
+        """Check that a valid full disk area is produced."""
+
+        good_area = AreaDefinition('gridded_himawari',
+                                   'A gridded Himawari area',
+                                   'longlat',
+                                   'EPSG:4326',
+                                   self.FULLDISK_SIZES[0.01]['x_size'],
+                                   self.FULLDISK_SIZES[0.01]['y_size'],
+                                   self.AHI_FULLDISK_EXTENT)
+
+        tmp_fh = self.make_fh('vis.01')
+        tmp_fh.get_area_def(None)
+        self.assertEqual(tmp_fh.area.area_extent, good_area.area_extent)
+        self.assertEqual(tmp_fh.area.area_id, good_area.area_id)
+        self.assertEqual(tmp_fh.area.description, good_area.description)
+        self.assertEqual(tmp_fh.area.area_extent, good_area.area_extent)
+        self.assertEqual(tmp_fh.area.proj_str, good_area.proj_str)
```

Comment:
Would it work to replace these with asserting the equality between `tmp_fh.area` and `good_area` ? (ie delegating the comparison to the area object)